app/src/processing/app/Base.java

```diff
@@ -1458,12 +1499,30 @@ public void actionPerformed(ActionEvent actionevent) {
     }));
     boardsCustomMenus.add(boardMenu);
 
+    // Insert recently used boards menu and remember index for insertion later
+    // Check if the field exists, in case preferences got lost
+    if (!PreferencesData.has("recent.num_boards")){
+        // (default to 5)
+          PreferencesData.setInteger("recent.num_boards", 5);
+    }
```

Comment:
I think that setting default values for preferences like this is not normally done, instead it should be added to `build/shared/lib/preferences.txt`, which is always loaded *before* the user's `preferences.txt` IIRC, so that sets the default value for when the user's file does not contain the preference (which is then saved to the user's `preferences.txt` on quit).

So this could just be removed.

```suggestion
```